crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -114,7 +124,23 @@ declare_lint! {
 
 declare_lint! {
     /// ## What it does
-    /// Checks whether an argument is used as both a value and a type form in a call
+    /// Checks whether an argument is used as both a value and a type form in a call.
+    ///
+    /// ## Why is this bad?
+    /// Such calls have confusing semantics and often indicate a logic error.
+    ///
+    /// ## Examples
+    /// ```python
+    /// from typing import reveal_type
+    /// from ty_extensions import is_fully_static
+    ///
+    /// if flag:
+    ///     f = repr  # Expects a value
+    /// else:
+    ///     f = is_fully_static  # Expects a type form
+    ///
+    /// reveal_type(f(int))  # str | Literal[True]
```

Comment:
I find the `reveal_type` call distracting here

```suggestion
    /// f(int)  # error
```